src/components/Identity/injectAwaitIdentityCookie.js

```diff
@@ -22,20 +30,32 @@ export default ({ orgId, doesIdentityCookieExist }) => {
         if (doesIdentityCookieExist()) {
           resolve();
         } else {
-          // This logic assumes that the code setting the cookie is working as expected and that
-          // the cookie was missing from the response.
-          const noIdentityCookieError = new Error(
-            `An identity was not set properly. Please verify that the org ID ${orgId} configured in Alloy matches the org ID specified in the edge configuration.`
-          );
+          let errorMessage;
+
+          const orgIdsFromCookies = extractOrgIdsFromCookies();
+          if (
+            orgIdsFromCookies.length > 0 &&
+            !includes(orgIdsFromCookies, orgId)
+          ) {
+            errorMessage = `An identity for organzation ${orgId} was not found. Valid organizations on this page are: ${orgIdsFromCookies.join(
+              ", "
+            )}`;
+          } else if (!domainMatchesApex(edgeDomain, apexDomain)) {
+            errorMessage = `An identity was not set properly because edge domain ${edgeDomain} does not match apex domain ${apexDomain}, and ${apexDomain} may not allow third-party cookies.`;
```

Comment:
We don't want to show this message when the customer is using third party data collection like adobedc.net, but I don't want to recreate the logic that Konductor uses to determine when to send cookies via cookie transfer and when to send them via set-cookie headers.